File: Estudiantes.API/Models/RegistroCurso.cs
using System.ComponentModel.DataAnnotations;

namespace Estudiantes.API.Models
{
    public class RegistroCurso
    {
        public int Id { get; set; }
        
        public Estudiante Estudiante { get; set; }
        
        public ICollection<Materia> Materias { get; set; }
        
        public Profesor Profesor { get; set; } 
        public Curso Curso { get; set; }
    }
}

File: EstudiantersFrontv1/src/app/models/registrocurso.ts
import { Curso } from "./curso";
import { Estudiante } from "./estudiante";
import { Materia } from "./materia";
import { Profesor } from "./profesor";



export interface RegistroCurso {
  id: number;
  estudiante: Estudiante;
  materias: Materia[];
  profesor: Profesor; 
  curso: Curso;
}